tap_google_ads/streams.py

```diff
@@ -330,22 +365,42 @@ def sync(self, sdk_client, customer, stream, config, state): # pylint: disable=u
         state = singer.set_currently_syncing(state, [stream_name, customer["customerId"]])
         singer.write_state(state)
 
-        query = create_core_stream_query(resource_name, selected_fields)
+        # last run was interrupted if there is a bookmark available for core streams.
+        last_pk_fetched = singer.get_bookmark(state,
+                                              stream["tap_stream_id"],
+                                              customer["customerId"]) or {}
+
+        # Assign True if the primary key is composite.
+        composite_pks = len(self.primary_keys) > 1
+
+        query = create_core_stream_query(resource_name, selected_fields, last_pk_fetched.get('last_pk_fetched'), self.filter_param, composite_pks)
```

Comment:
`create_core_stream_query` is an existing function. They kept query_building and other helper functions out of class because `Basestream` is the parent class of `UserInterestStream` and `ReportStream`.  `create query`([link](https://github.com/singer-io/tap-google-ads/blob/TDL-18749-Implement-Interruptible-Full-Table/tap_google_ads/streams.py#L129)) function for report stream is also placed out of `BaseStream` class. So, I think it would be fine to not move the `create query` function in BaseStream class.